File: Week 4/templates/C_horrorlist.cpp
#include <bits/stdc++.h>
using namespace std;

//input format: related[i] is a list of all indices related to i
//return the integer that has the highest horror index
int horrorlist(int n, vector<int>& horror_movies, vector<vector<int>>& related){
    // Your code goes here
    return -1; // This is a placeholder
}
int main(){
    int n, h, l;
    cin >> n >> h >> l;
    vector<int> horror_movies(h);
    for(int i = 0; i < h; i++){
        cin >> horror_movies[i];
    }
    vector<vector<int>> edges(n);
    for(int i = 0; i < l; i++){
        int u, v;
        cin >> u >> v;
        edges[u].push_back(v);
        edges[v].push_back(u);
    }
    cout << horrorlist(n, horror_movies, edges) << endl;
}

File: Week4/templates/C_horrorlist.py
from queue import Queue

INF = 10 ** 9 # useful way to keep track of infinity as a really large number

#input format: for each index i, related[i] is a list of indices related to i
def horrorlist(n, horror_movies, related):
    # Your code to find the index with the highest horror-index goes here
    return None # This is a placeholder, replace it with your code


n,h,l = map(int, input().split())
horror_movies = list(map(int, input().split()))
related = [ [] for i in range(n) ]
for i in range(l):
    u,v = map(int, input().split())
    related[u].append(v)
    related[v].append(u)

print(horrorlist(n, horror_movies, related))